packages/camera/camera_windows/example/lib/main.dart

```diff
@@ -0,0 +1,417 @@
+    assert(_cameraId > 0);
+    assert(_initialized);
+    try {
+      await CameraPlatform.instance.dispose(_cameraId);
+      setState(() {
+        _initialized = false;
+        _cameraId = -1;
+        _cameraInfo = 'Camera disposed';
+        _previewSize = null;
+        _recording = false;
+        _recordingTimed = false;
+        _previewPaused = false;
+      });
+      await _getAvailableCameras();
+    } on CameraException catch (e) {
+      setState(() {
+        _cameraInfo = 'Failed to dispose camera: ${e.code}: ${e.description}';
+      });
+    }
+  }
+
+  Widget _buildPreview() {
+    return CameraPlatform.instance.buildPreview(_cameraId);
+  }
+
+  Future<void> _takePicture() async {
+    final XFile _file = await CameraPlatform.instance.takePicture(_cameraId);
+    _showInSnackBar('Picture captured to: ${_file.path}');
+  }
+
+  Future<void> _recordTimed(int seconds) async {
+    if (_initialized && _cameraId > 0 && !_recordingTimed) {
+      CameraPlatform.instance
+          .onVideoRecordedEvent(_cameraId)
+          .first
+          .then((VideoRecordedEvent event) async {
+        if (mounted) {
+          setState(() {
+            _recordingTimed = false;
+          });
+
+          _showInSnackBar('Video captured to: ${event.file.path}');
+        }
+      });
+
+      await CameraPlatform.instance.startVideoRecording(
+        _cameraId,
+        maxVideoDuration: Duration(seconds: seconds),
+      );
+
+      setState(() {
+        _recordingTimed = true;
+      });
+    }
+  }
+
+  Future<void> _toggleRecord() async {
+    if (_initialized && _cameraId > 0) {
+      if (_recordingTimed) {
+        /// Request to stop timed recording short.
+        await CameraPlatform.instance.stopVideoRecording(_cameraId);
+      } else {
+        if (!_recording) {
+          await CameraPlatform.instance.startVideoRecording(_cameraId);
+        } else {
+          final XFile _file =
+              await CameraPlatform.instance.stopVideoRecording(_cameraId);
+
+          _showInSnackBar('Video captured to: ${_file.path}');
+        }
+        setState(() {
+          _recording = !_recording;
+        });
+      }
+    }
+  }
+
+  Future<void> _togglePreview() async {
+    if (_initialized && _cameraId >= 0) {
+      if (!_previewPaused) {
+        await CameraPlatform.instance.pausePreview(_cameraId);
+      } else {
+        await CameraPlatform.instance.resumePreview(_cameraId);
+      }
+      setState(() {
+        _previewPaused = !_previewPaused;
+      });
+    }
+  }
+
+  Future<void> _switchCamera() async {
+    _cameraIndex = (_cameraIndex + 1) % _cameras.length;
+    if (_initialized && _cameraId >= 0) {
+      await _disposeCurrentCamera();
+      if (_cameras.isNotEmpty) {
+        await _initializeCamera();
+      }
+    } else {
+      await _getAvailableCameras();
+    }
+  }
+
+  Future<void> _onResolutionChange(ResolutionPreset? newValue) async {
+    if (newValue == null) {
+      return;
+    }
```

Comment:
It seems like null resolution presets are nonsensical here. Consider declaring `newValue` non-null in the parameters list and not calling this method when we have null values. That keeps all the null handling near the outer API surface of this plugin rather than handling them throughout.